### PR TITLE
renewal of the shiny interface

### PR DESCRIPTION
**Pull request by jkeuskamp**
To facilitate renewal of the shiny interface two small changes were made to increase loading speed and predictability of thumbnails.

### CODE REVIEW

File: R/locate_area.R [hunk 1]
```diff
@@ -53,9 +54,16 @@ function(file, pages = NULL, resolution = 60L, widget = c("shiny", "native", "re
     file <- localize_file(file, copy = TRUE)
     on.exit(unlink(file), add = TRUE)
     dims <- get_page_dims(file, pages = pages)
+   
+if (!is.null(thumbnails)){
```
Comment (leeper) [inline R/locate_area.R]: Add space between `){`.

File: R/locate_area.R [hunk 2]
```diff
@@ -53,9 +54,16 @@ function(file, pages = NULL, resolution = 60L, widget = c("shiny", "native", "re
...
+    filelist<-list.files(path.expand(thumbnails), pattern = "\\.png$", ignore.case=TRUE, full.names=TRUE)
+    file.copy(filelist, tempdir(),overwrite=T)
+    paths <-file.path(tempdir(),basename(filelist))
+    cat('fetching files')
+} else if (is.null(thumbnails)){
```
Comment (leeper) [inline R/locate_area.R]: Add space between `){`.

File: R/locate_area.R [hunk 3]
```diff
@@ -97,7 +105,7 @@ function(file, pages = NULL, resolution = 60L, widget = c("shiny", "native", "re
 #' @rdname extract_areas
 #' @export
 extract_areas <- function(file, pages = NULL, guess = FALSE, ...) {
-    areas <- locate_areas(file = file, pages = pages)
+    areas <- locate_areas(file = file, pages = pages, thumbnails = NULL, resolution = 60L, widget = c("shiny", "native", "reduced"))
```
Comment (leeper) [inline R/locate_area.R]: Why did this line change?

File: R/make_thumbnails.R [hunk 4]
```diff
@@ -48,6 +49,6 @@ function(file, outdir = NULL, pages = NULL, format = c("png", "jpeg", "bmp", "gi
         PDFImageWriter$writeImage(pdfDocument, format, "", pages[i], pages[i], 
                                   prefix, 1L, as.integer(resolution))
     }
-    file.rename(from = paste0(prefix, pages, ".", format), to = outfile)
-    ifelse(file.exists(outfile), outfile, NA_character_)
+    if(isTRUE(uniform)){file.rename(from = paste0(prefix, pages, ".", format), to = outfile)
```
Comment (leeper) [inline R/make_thumbnails.R]: Spread this out across multiple lines with spacing between brackets:

```R
if (condition) {
 code
}
```

File: R/make_thumbnails.R [hunk 5]
```diff
@@ -6,6 +6,7 @@
 #' @param format A character string specifying an image file format.
 #' @param resolution An integer specifying the image resolution in DPI.
 #' @param password Optionally, a character string containing a user password to access a secured PDF.
+#' @param uniform determines whether the page numbers suffixes of generated thumbnails have a uniform (01,02,...,10) or short (1,2,...,10) format.
```
Comment (leeper) [inline R/make_thumbnails.R]: Capitalize "Determines"

**Comment by pachadotdev** [comment]
I will need to check the changes with tabula 1.0.5

**Comment by pachadotdev** [comment]
thanks @jkeuskamp

I added parts of it manually after multiple changes to the codebase

**Pull request closed without merging**